src/compiler/checker.ts

```diff
@@ -4533,6 +4551,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 symbol = getMergedSymbol(getSymbol(getExportsOfSymbol(resolveAlias(namespace)), right.escapedText, meaning));
             }
             if (!symbol) {
+                if (isExportAssignmentAny(namespace)) {
+                    return unknownSymbol; // no error - lookup on psuedo-`any` module

```

Comment:
Previously this was handled one level up the stack because the module itself was the `any`-typed `const`. Now it's not `any` (and can contain other well-typed members, like `@typedef`s), but it contains an `export=` that points at `any`, and we check that there to reenable the error-less lookup it used to have.